File: 2D_amber/Assets/Scripts/JS/Map/Spawner.js
#pragma strict
var target : GameObject ;
var number : int = 1;
var delay : float = 0.1 ;

private var box : Collider2D ;

function Awake(){
  box = GetComponent.<Collider2D>() ;
}

function Spawn(){
  var max : Vector3 = box.bounds.max ;
  var min : Vector3 = box.bounds.min ;
  var pos : Vector3 = transform.position ;
  pos.x = Random.Range(min.x, max.x) ;
  
  Instantiate(target, pos, transform.rotation) ;
}

function DelaySpawn(){
  for(var i=0 ; i<number ; i++){
    Spawn() ;
    yield WaitForSeconds(delay) ;
  }
}

function CallSpawn(){
  DelaySpawn() ;
}

File: 2D_amber/Assets/Scripts/CSharp/Map/Spawner.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Spawner : MonoBehaviour {

	public GameObject target ;
	public int number = 1 ;
	public float delay = 0.1f ;

	Collider2D box ;

	void Awake(){
		box = GetComponent<Collider2D>() ;
	}

	public void Spawn(){
		Vector3 max = box.bounds.max ;
		Vector3 min = box.bounds.min ;
		Vector3 pos = transform.position ;
		pos.x = Random.Range(min.x, max.x) ;
  
		Instantiate(target, pos, transform.rotation) ;
	}

	public IEnumerator DelaySpawn(){
		for(var i=0 ; i<number ; i++){
			Spawn() ;
			yield return new WaitForSeconds(delay) ;
		}
	}

	public void CallSpawn(){
		DelaySpawn() ;
	}
}
